packages/clerk-js/src/ui/components/OrganizationProfile/InviteMembersForm.tsx

```diff
@@ -89,24 +89,34 @@ export const InviteMembersForm = (props: InviteMembersFormProps) => {
       .inviteMembers({ emailAddresses: emailAddressField.value.split(','), role: roleField.value as MembershipRole })
       .then(onSuccess)
       .catch(err => {
+        removeInvalidEmails(err);
+
         if (isClerkAPIResponseError(err) && err.errors?.[0]?.code === 'duplicate_record') {
           const unlocalizedEmailsList = err.errors[0].meta?.emailAddresses || [];
 
           // Create a localized list of email addresses
           const localizedList = createListFormat(unlocalizedEmailsList, locale);
           setLocalizedEmails(localizedList);
-
-          // Remove any invalid email address
-          const invalids = new Set(unlocalizedEmailsList);
-          const emails = emailAddressField.value.split(',');
-          emailAddressField.setValue(emails.filter(e => !invalids.has(e)).join(','));
         } else {
           setLocalizedEmails(null);
           handleError(err, [], card.setError);
         }
       });
   };
 
+  const removeInvalidEmails = (err: any) => {
+    if (!isClerkAPIResponseError(err)) {
```

Comment:
❓ the removal of invalid email addresses was executed only when the error code was `duplicate_code` but after this change, it's executed for all error codes. Is this intentional?

Reply:
invalid email addresses will be return in `meta.emailAddresses` and we want to remove them from the text field despite the error code returned

Reply:
We would like to remove any email addresses included in the `meta.emailAddresses` & `meta.identifiers` properties regardless the error code